lib/auth/new_web_user.go

```diff
@@ -291,6 +315,14 @@ func (s *AuthServer) CreateUserWithoutOTP(token string, password string) (servic
 }
 
 func (s *AuthServer) CreateUserWithU2FToken(token string, password string, response u2f.RegisterResponse) (services.WebSession, error) {
```

Comment:
Instead of checking for this condition in all these methods:

```
CreateSignupToken
CreateUserWithU2FToken
CreateUserWithOTP
CreateUserWithoutOTP
```

We can do it in 1 method, say `CreateUser`.

This is how we do it in gravity https://github.com/gravitational/gravity/blob/master/lib/users/usersservice/usersservice.go#L1340
where all requests for creating a new local user go through 1 method which has all the checks.

Reply:
I kind of agree that having one place is nice, not sure if worth refactoring it all right now makes sense